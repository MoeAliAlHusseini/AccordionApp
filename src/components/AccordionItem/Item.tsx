/* eslint-disable prettier/prettier */
import React from 'react';

import { View, Text, TouchableOpacity, Image } from 'react-native';

import styles from './Item.component.style';
import Colors from '../../style/colors';
import Icons from '../../style/icons';

interface ItemProps {
    id: number;
    title: string;
    description: string;
    experience: string;
    isOpen: boolean;
    onItemPress(id: number): void
    onLearnMorePress(id: number): void
}

const Item = (props: ItemProps) => {
    return (
        <>
            <View style={styles.container}>
                <TouchableOpacity testID="itemTouchableOpacity" activeOpacity={0.6} key={`item - ${props.id}`} style={styles.touchableOpacityContainer} onPress={() => props.onItemPress(props.id)}>
                    <View>
                        <Text style={styles.title}>Name: {props.title}</Text>
                        <Text style={styles.description}>Job: {props.description}</Text>
                    </View>
                    <Image source={Icons.chevronIcon} style={[styles.chevron, { transform: [{ rotate: props.isOpen ? '0deg' : '180deg' }] }]} />
                </TouchableOpacity>
                {props.isOpen && <View>
                    <Text>
                        {props.experience}
                    </Text>

                </View>}
                {props.isOpen && <TouchableOpacity testID="learnMoreTouchableOpacity" onPress={() => props.onLearnMorePress(props.id)}><Text style={{ color: Colors.BLUE }}>Learn more</Text></TouchableOpacity>}
            </View>

        </>
    );
};

export default Item;
