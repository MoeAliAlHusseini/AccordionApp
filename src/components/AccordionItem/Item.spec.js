import * as React from 'react';
import {render, screen} from '@testing-library/react-native';

import Item from './Item';

describe('Item', () => {
  let props = {
    id: 0,
    title: 'Engineer',
    description: 'FrontEnd',
    experience: '5 years',
    isOpen: false,
    onItemPress: jest.fn(),
    onLearnMorePress: jest.fn(),
  };

  beforeEach(() => {
    render(<Item props={props} />);
  });

  it('should render', () => {
    expect(screen).toMatchSnapshot();

    screen.debug();
  });

  it('should check the text is rendered', () => {
    let nameText = screen.getByText('Name:');
    expect(nameText).toBeDefined();
  });
});
