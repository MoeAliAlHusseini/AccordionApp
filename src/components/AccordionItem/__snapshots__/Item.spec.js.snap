// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render 1`] = `
<View
  style={
    {
      "borderColor": "green",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "column",
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexDirection": "row",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          {
            "marginBottom": 20,
          }
        }
      >
        Name: 
      </Text>
      <Text
        style={
          {
            "marginBottom": 10,
          }
        }
      >
        Job: 
      </Text>
    </View>
    <Image
      source={
        {
          "testUri": "../../../src/style/icons/chevron.png",
        }
      }
      style={
        [
          {
            "height": 12,
            "marginLeft": "auto",
            "width": 12,
          },
          {
            "transform": [
              {
                "rotate": "180deg",
              },
            ],
          },
        ]
      }
    />
  </View>
</View>
`;
